src/postgraphql/schema/procedures/createPgProcedureQueryGqlFieldEntry.ts

```diff
@@ -51,12 +52,17 @@ function createPgSingleProcedureQueryGqlFieldEntry (
     type: fixtures.return.gqlType,
     args: buildObject(argEntries),
 
-    async resolve (_source, args, context): Promise<mixed> {
+    async resolve (_source, args, context, resolveInfo): Promise<mixed> {
       const client = pgClientFromContext(context)
       const input = argEntries.map(([argName], i) => fixtures.args[i].fromGqlInput(args[argName]))
-      const query = sql.compile(sql.query`select to_json(${createPgProcedureSqlCall(fixtures, input)}) as value`)
+      const aliasIdentifier = Symbol()
+      const query = sql.compile(sql.query`
+        select ${getSelectFragment(resolveInfo, aliasIdentifier, fixtures.return.gqlType)} as value,
+        (${sql.identifier(aliasIdentifier)} is null) as is_null
+        from ${createPgProcedureSqlCall(fixtures, input)} as ${sql.identifier(aliasIdentifier)}
+      `)
       const { rows: [row] } = await client.query(query)
-      return row ? fixtures.return.intoGqlOutput(fixtures.return.type.transformPgValueIntoValue(row['value'])) : null
+      return row && !row['is_null'] ? fixtures.return.intoGqlOutput(fixtures.return.type.transformPgValueIntoValue(row['value'])) : null
     },
   }]
 }
```

Comment:
Basically I've not found a way to work around the `with` query forcing a column alias. I think in the event of returning scalars (which we know up front, right?) we can just return the only value in a single-keyed object?